confgenerator/otel/processors.go

```diff
@@ -386,3 +386,32 @@ func GCPResourceDetector(override bool) Component {
 		Config: config,
 	}
 }
+
+// ResourceTransform returns a Component that applies changes on resource attributes.
+func ResourceTransform(attributes map[string]string, GCPResourceDetector bool) Component {
```

Comment:
I think the bool parameter can just be called something like `override bool`. I don't understand why you've called it `GCPResourceDetector`?